contracts/6/forwarder/ERC20Forwarder.sol

```diff
@@ -0,0 +1,264 @@
+ * @dev Tx Flow : calls BiconomyForwarder to handle forwarding, call _transferHandler() to charge fee after
+ *
+ */
+ contract ERC20Forwarder is ERC20ForwarderStorage {
+     using SafeMath for uint256;
+     using SafeMath for uint128;
+     bool internal initialized;
+      
+     /**
+     * @dev sets contract variables
+     *
+     * @param _feeReceiver : address that will receive fees charged in ERC20 tokens
+     * @param _feeManager : the address of the contract that controls the charging of fees
+     * @param _forwarder : the address of the BiconomyForwarder contract
+     *
+     */
+       function initialize(
+       address _feeReceiver,
+       address _feeManager,
+       address payable _forwarder
+       ) public {
+        
+        require(!initialized, "ERC20 Forwarder: contract is already initialized");
+        require(
+            _feeReceiver != address(0),
+            "ERC20Forwarder: new fee receiver is the zero address"
+        );
+        require(
+            _feeManager != address(0),
+            "ERC20Forwarder: new fee manager is the zero address"
+        );
+        require(
+            _forwarder != address(0),
+            "ERC20Forwarder: new forwarder is the zero address"
+        );
+        initialized = true;
+        feeReceiver = _feeReceiver;
+        feeManager = _feeManager;
+        forwarder = _forwarder;
+       }
+       
+    
+    function setOracleAggregator(address oa) external onlyOwner{
+        oracleAggregator = oa;
+    }
+
+
+    function setBaseGas(uint128 gas) external onlyOwner{
+        baseGas = gas;
+    }
+    
+    function setGasRefund(uint128 refund) external onlyOwner{
+        gasRefund = refund;
+    }
+
+    /**
+     * @dev enable dApps to change fee receiver addresses, e.g. for rotating keys/security purposes
+     * @param _feeReceiver : address that will receive fees charged in ERC20 tokens */
+    function setFeeReceiver(address _feeReceiver) external onlyOwner{
+        feeReceiver = _feeReceiver;
+    }
+    
+    
+    /**
+     * @dev enable dApps to change the contract that manages fee collection logic
+     * @param _feeManager : the address of the contract that controls the charging of fees */
+    function setFeeManager(address _feeManager) external onlyOwner{
+        feeManager = _feeManager;
+    }
+    
+    
+    /**
+     * @dev change amount of excess gas charged for _transferHandler
+     * NOT INTENTED TO BE CALLED : may need to be called if :
+     * - new feeManager consumes more/less gas
+     * - token contract is upgraded to consume less gas
+     * - etc
+     */
+     /// @param _transferHandlerGas : max amount of gas the function _transferHandler is expected to use
+    function setTransferHandlerGas(address token, uint256 _transferHandlerGas) external onlyOwner{
+        transferHandlerGas[token] = _transferHandlerGas;
```

Comment:
Emit an event in case of transfer handler gas change.. so community members can track this event emission for any change we do in our contracts.. same goes for when changing the baseGas fee or gas refund

Reply:
ok three events added